### PR TITLE
add check format providerId

### PR DESCRIPTION
**Pull request by outscale-hmi**
**What type of PR is this?**
> /kind bug
 
**Which issue(s) this PR fixes**:
*Incorrect ReadVms syntax. https://github.com/outscale/cloud-provider-osc/issues/348

### CODE REVIEW

File: cloud-controller-manager/osc/instances_v2.go [hunk 1]
```diff
@@ -245,6 +245,11 @@ func parseInstanceIDFromProviderIDV2(providerID string) (string, error) {
 	// * /<availability-zone>/<instance-id>
 	// * <instance-id>
```
Comment (outscale-mdr) [inline cloud-controller-manager/osc/instances_v2.go:246]: If we want to still keep the possibility to set only the instance Id, check if the instance-id is `i-*`

**Review by outscale-mdr — CHANGES_REQUESTED** [review]

File: cloud-controller-manager/osc/instances_v2.go [hunk 2]
```diff
@@ -245,6 +245,15 @@ func parseInstanceIDFromProviderIDV2(providerID string) (string, error) {
...
 	instanceID := ""
+	if !strings.HasPrefix(providerID, "i-") && !strings.HasPrefix(providerID, "aws://") {
+	// providerID does not have the expected prefix
+		return "", errors.New("invalid providerID format: missing 'aws://' prefix or 'i-' prefix")
+	}
+	// Handle case where providerID is in format 'i-<instance-id>'
+	if strings.HasPrefix(providerID, "i-") {
+		return providerID, nil
+	}
```
Comment (outscale-mdr) [inline cloud-controller-manager/osc/instances_v2.go:255]: I think the [TrimPrefix](https://pkg.go.dev/strings#TrimPrefix) line 257 trims if it exist or it does nothing therefore the lines 252-255 are already handled by the line 258 `if len(metadata) == 1 {`.

Do you agree ?

**Review by outscale-mdr — CHANGES_REQUESTED** [review]

File: cloud-controller-manager/osc/instances_v2.go [hunk 3]
```diff
@@ -255,6 +264,11 @@ func parseInstanceIDFromProviderIDV2(providerID string) (string, error) {
 	} else if len(metadata) == 3 {
 		// /az/instance-id
 		instanceID = metadata[2]
+	} else {
+		return "", errors.New("invalid providerID format")
+	}
+	if instanceID == "" {
+		return "", fmt.Errorf("instance ID not found in providerID: %s", providerID)
```
Comment (outscale-mdr) [inline cloud-controller-manager/osc/instances_v2.go:271]: Unreachable code with the `else`, no ?

File: cloud-controller-manager/osc/instances_v2.go (hunk 2 above)
Comment (outscale-hmi) [reply cloud-controller-manager/osc/instances_v2.go:255]: yes, you're right, 
If the prefix doesn't exist in the string, strings.TrimPrefix returns the original string unchanged, so" i-providerId" should be handled any ways

File: cloud-controller-manager/osc/instances_v2.go (hunk 3 above)
Comment (outscale-hmi) [reply cloud-controller-manager/osc/instances_v2.go:271]: what if  providerId "aws://":
strings.TrimPrefix(providerID, "aws://") results in "" 
=> len(metadata) == 1 => instanceID = metadata[0]="" => so instanceId will not be found 
Am I right ?

Comment (outscale-mdr) [reply cloud-controller-manager/osc/instances_v2.go:271]: You're right, the providerId needs to be check in all cases

**Review by outscale-mdr — APPROVED** [review]
Good work @outscale-hmi

**Pull request merged**